### PR TITLE
Hooked up the Models with http client

### PR DESCRIPTION
**Pull request by MehrdadMzfr**
Using model object for token result

Things to consider before you submit the PR:
* [X ] Are tests passing locally?
* [ X] Are the files formatted correctly?
* [ X] Did you check UI tests on the sample app? They are not executed on CI.
* [X ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 1]
```diff
@@ -18,8 +22,19 @@ + (void)tokenAsync:(MSStorageIngestion *)httpClient
...
-        NSLog(@"Get token callback, request Id %@ with status code: %lu", callId, (unsigned long)response.statusCode);
-        completion(data, error);
+        MSLogVerbose([MSDataStorage logTag], @"Get token callback, request Id %@ with status code: %lu", callId, (unsigned long)response.statusCode);
+
+        // Read tokens.
+        NSError *tokenResponsejsonError;
+        NSDictionary *jsonDictionary= [NSJSONSerialization JSONObjectWithData:data options:0 error:&tokenResponsejsonError];
+        if (tokenResponsejsonError) {
+          MSLogError([MSDataStorage logTag], @"Can't deserialize tokens with error: %@", [tokenResponsejsonError description]);
+          completion([[MSTokensResponse alloc] initWithTokens:nil], error);
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: Can `completion` be`nil`?

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Model/MSTokensResponse.m [hunk 2]
```diff
@@ -12,4 +22,27 @@ - (instancetype)initWithTokens:(NSArray<MSTokenResult *> *)tokens {
   return self;
 }
 
+- (instancetype)initWithDictionary:(NSDictionary *)tokens {
+  
+  self = [super init];
+  if (self) {
+    NSInteger count = [(NSArray *)tokens[kMSTokens] count];
+    if ( count > 0) {
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Model/MSTokensResponse.m]:
```suggestion
    if (count > 0) {
```

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Model/MSTokensResponse.m [hunk 3]
```diff
@@ -12,4 +22,27 @@ - (instancetype)initWithTokens:(NSArray<MSTokenResult *> *)tokens {
...
+      NSMutableArray *tokenArray = [[NSMutableArray alloc] initWithCapacity:count];
+      for (NSDictionary *dic in tokens[kMSTokens]) {
+        MSTokenResult *result = [[MSTokenResult alloc]
+                                 initWithPartition:dic[kMSPartition]
+                                 dbAccount:dic[kMSDbAccount]
+                                 dbName:dic[kMSDbName]
+                                 dbCollectionName:dic[kMSDbCollectionName]
+                                 token:dic[kMSToken]
+                                 status:dic[kMSStatus]];
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Model/MSTokensResponse.m]: Please format

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.h [hunk 4]
```diff
@@ -2,7 +2,8 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-typedef void (^MSGetTokenAsyncCompletionHandler)(NSData *data, NSError *error);
+@class MSTokensResponse;
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.h]: Group forward declarations into one block

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 5]
```diff
@@ -1,5 +1,9 @@
 #import "MSTokenExchange.h"
 #import "MSStorageIngestion.h"
+#import "MSTokensResponse.h"
+#import "AppCenter+Internal.h"
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: nitpick: sort + (system ones always come first)

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m (hunk 1 above)
Comment (MehrdadMzfr) [reply AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]: added check

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 6]
```diff
@@ -18,8 +21,22 @@ + (void)tokenAsync:(MSStorageIngestion *)httpClient
...
+        // If comletion is provided.
+        if (completion) {
+          // Read tokens.
+          NSError *tokenResponsejsonError;
+          NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&tokenResponsejsonError];
+          if (tokenResponsejsonError) {
+            MSLogError([MSDataStorage logTag], @"Can't deserialize tokens with error: %@", [tokenResponsejsonError description]);
+            completion([[MSTokensResponse alloc] initWithTokens:nil], error);
+          }
+          // Create token result object
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]:
```suggestion

          // Create token result object.
```

File: AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m [hunk 7]
```diff
@@ -18,8 +21,22 @@ + (void)tokenAsync:(MSStorageIngestion *)httpClient
...
   [httpClient sendAsync:payloadData
       completionHandler:^(NSString *callId, NSHTTPURLResponse *response, NSData *data, NSError *error) {
-        NSLog(@"Get token callback, request Id %@ with status code: %lu", callId, (unsigned long)response.statusCode);
-        completion(data, error);
+        MSLogVerbose([MSDataStorage logTag], @"Get token callback, request Id %@ with status code: %lu", callId,
+                     (unsigned long)response.statusCode);
+
+        // If comletion is provided.
+        if (completion) {
+          // Read tokens.
```
Comment (MatkovIvan) [inline AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m]:
```suggestion

          // Read tokens.
```

**Pull request merged**